### PR TITLE
TT-9527-enable otel for response middlewares

### PR DESCRIPTION
**Pull request by sredxny**
## Description

Added support for otel in response middlewares

## Related Issue

TT-9527

## Motivation and Context

Enable tracing for response middlewares

## How This Has Been Tested

- Added tests with tracetest
- Manually executed the gateway and then:
    -  added an api with header injector enabled
    - added an api without response middlewares enabled
    - added an api with ResponseGoPlugin loaded
    - Send traffic to each api and we must see the spans for the required response middlewares in jaegger

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] I ensured that the documentation is up to date
- [ ] I explained why this PR updates go.mod in detail with reasoning why it's required
- [ ] I would like a code coverage CI quality gate exception and have explained why

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5389/merge
Commit: d75d243b846431e76c97c90c1a6abbf7d0065a41 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5766300805)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5389/merge
Commit:  
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5766181140)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5389/merge
Commit: 360622c0be66373d85fbe129bd617b1ed284229e 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5773037246)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5389/merge
Commit: bbcc89eb717dea849368b2939082b570125c8828 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5790550852)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5389/merge
Commit: 20b45e162aadd2a4b8158006c8362087775f2aa4 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5790568789)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5389/merge
Commit: 41fb1113b293f8e1ca9f03dfdf9a32b2f2f6986e 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5792111275)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5389/merge
Commit: 41fb1113b293f8e1ca9f03dfdf9a32b2f2f6986e 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5792111275)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5389/merge
Commit: 939c993bccbbcbcbfb8625779529be8c77c462a5 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5792501627)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5389/merge
Commit: cd454684809cb9b63ffa470d50575b4e3de6b1e2 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5792532111)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5389/merge
Commit: 5cceb4775fa76a3fa22f4af2b9faa4ec5fadb1fb 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5803480838)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5389/merge
Commit: 1644c7cecd1f741d3d85f8f40659335c26d88046 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5803490699)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5389/merge
Commit: e1852c3c9643e35eafb4703648a30a9ef1cd7b69 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5803566019)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5389/merge
Commit: ecd82c9d2055c581fc23e2fa4e28b96c7ac21d8a 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5804495581)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5389/merge
Commit: 0c2bfd864cdcdf42a33e3c0e59f5b73fb74f1496 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5810746715)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5389/merge
Commit: 13a5bb3a3322e74dfb6e79177d12ec6907a98c6a 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5810931206)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5389/merge
Commit: 2d7606f1a19d3aaa65851bd0f63db39d6f03d8c6 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5812131040)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5389/merge
Commit: 8cce742c3ccbcbeb61d430823bef0664c7f7f8d5 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5812387632)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5389/merge
Commit: cba7873fa88467dce16a59ef71e9a1769faf68f3 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5812426525)

File: ci/tests/tracing/scenarios/tyk_test_with_response_mw.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  description: test the execution of the response mw
+  trigger:
+    type: http
+    httpRequest:
+      method: GET
+      url: tyk:8080/test-response-mw/ip
+      headers:
+        - key: Content-Type
+          value: application/json
+  specs:
```
Comment (mativm02) [inline ci/tests/tracing/scenarios/tyk_test_with_response_mw.yml:14]: can you check here if this span contains the actual parent span? to check the issue we discussed earlier in the Slack channel - I don't think it'd be so difficult with tracetest

**Review by mativm02 — CHANGES_REQUESTED** [review]

File: ci/tests/tracing/scenarios/tyk_test_with_response_mw.yml (hunk 1 above)
Comment (sredxny) [reply ci/tests/tracing/scenarios/tyk_test_with_response_mw.yml:14]: makes the whole sense, done!

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5389/merge
Commit: 33ddd0f9756589e7f97516c664fdf69e0451a1ac 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5814417811)

**Review by mativm02 — APPROVED** [review]
lgtm!

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5389/merge
Commit: 3246a325fadf0bd07a5649a5ab149c98c7216c8c 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5824162059)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5389/merge
Commit: c264d3ffc293926ac124310092f4543df524f8cc 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5824177168)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5389/merge
Commit: 2c621f9ad374065f6a94131de59b627783b43411 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5825460836)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5389/merge
Commit: 350ad92fc5ce9928add4139cac5468fbcb9aae1d 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5828119949)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5389/merge
Commit: d085b1a5e04be96f519b5aead5a93df0b8632cdd 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5828300540)

**Comment by sredxny** [comment]
SonarCloud failing in lines that were not touched in this PR...also tests were added using tracetests

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5389/merge
Commit: 455779e30059e91984dd5ffd6b5db07ee09125f1 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5828350191)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5389/merge
Commit: 938e69cb3342a3ce2b36102f34f3366a1b10dfb1 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5828406376)

**Comment by sredxny** [comment]
Sonar cloud is complaining for some lines that were not touched in this PR, also tests were added using tracetest

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5389/merge
Commit: ef95c7a7c43dcc4de2240211238561b686c66ddf 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5828466056)

**Review by tbuchaillot — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5389/merge
Commit: bc8e77180578dc70ebb3796dc8e1131026b02c7d 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5832448805)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5389/merge
Commit: f9be40d7ddb8c4f37e19d593832c93a85cc6d3f6 
Triggered by: pull_request (@mativm02)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5832495780)

**Pull request merged**